tests/testthat/test-process_repos.R

```diff
@@ -0,0 +1,38 @@
+test_that("process_repos() returns results with repo metadata", {
+
+  my_project <- 
+    initialize_project("gitai_test_project") |> 
+    set_github_repos(
+      repos = c("r-world-devs/GitStats", "openpharma/DataFakeR")
+    ) |>
+    add_files(file_paths = "README.md") |>
```

Comment:
Maciej, regarding this add_files() maybe we need then two `add_file_paths()` and `add_file_types()` ?

Reply:
For the sake of simplicity maybe let's leave `add_files()` with one parameter where we can pass a `file type` or a `file path`? For the time being it would be treated by `GitStats` similarly (with `get_files_structure()` and `get_files_content()`) and in future we could think of making regex distinction already in code (not by user then, it would be done automatically) which would then create efficiency gains?

Reply:
Yes, I like this solution! I think that's what I was looking for. Let's leave it in this branch - I will finish it first. And once merged feel free to refactored the add_files() function.